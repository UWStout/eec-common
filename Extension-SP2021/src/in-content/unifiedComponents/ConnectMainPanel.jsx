import React, { useState } from 'react'
import PropTypes from 'prop-types'

import { Paper, Typography } from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'

import PanelTitle from './PanelTitle.jsx'
import ConnectMainContent from './ConnectMainContent.jsx'

// Just for dummy content while testing
import { LoremIpsum } from 'lorem-ipsum'
const lorem = new LoremIpsum({
  sentencesPerParagraph: {
    max: 8,
    min: 4
  },
  wordsPerSentence: {
    max: 16,
    min: 4
  }
})

const useStyles = makeStyles((theme) => ({
  // Styling of the root paper element
  paperRoot: {
    // Margins and sizing
    padding: theme.spacing(2),
    paddingRight: theme.spacing(4),
    width: theme.spacing(25),
    height: theme.spacing(60),

    // Position element in window
    position: 'fixed',
    top: '20%',

    // Animate changes in the 'right' property
    transition: theme.transitions.create(
      ['right'], { duration: theme.transitions.duration.standard }
    )
  },

  // Style when the panel is retracted
  panelRetracted: {
    right: -theme.spacing(16)
  },

  // Style when the panel is fully expanded
  panelExpanded: {
    right: -theme.spacing(2)
  },

  // Style when the panel is hidden
  panelHidden: {
    right: -theme.spacing(35)
  },

  // Styling for the outer box
  boxStyle: {
    height: theme.spacing(60),
    paddingTop: theme.spacing(5),
    marginLeft: '-2px',
    marginRight: '-7px'
  },

  // Styling for the primary content box
  contentStyle: {
    height: theme.spacing(54),
    overflowY: 'auto'
  },

  paragraph: {
    marginBottom: theme.spacing(2)
  }
}))

/**
 * The main Karuna Connect panel showing info about user status, team status,
 * and more with functionality to update status.
 *
 * @param {object} props Properties for the component (See propTypes)
 * @returns {React.Element} The element to render for this component
 */
export default function ConnectMainPanel ({ hidden, onHide, waitToHide }) {
  // Deconstruct props and style class names
  const {
    paperRoot,
    panelHidden,
    panelRetracted,
    panelExpanded,
    contentStyle,
    boxStyle,
    paragraph
  } = useStyles()

  // Hover state of mouse
  const [mouseIsOver, setMouseIsOver] = useState(false)

  // Handle to the pending hide request (if any)
  const [hideTimeout, setHideTimeout] = useState(false)

  const [paragraphs, setParagraphs] = useState(['', ''])
  React.useEffect(() => {
    setParagraphs([
      lorem.generateParagraphs(1),
      lorem.generateParagraphs(1)
    ])
  }, [])

  // Function for queueing a hide request
  const hide = (immediate) => {
    if (onHide && !hidden) {
      if (immediate) {
        onHide()
      } else {
        const timeoutHandle = setTimeout(() => { onHide() }, waitToHide)
        setHideTimeout(timeoutHandle)
      }
    }
  }

  // Function for canceling a pending hide request
  const cancelHide = () => {
    if (hideTimeout) {
      clearTimeout(hideTimeout)
      setHideTimeout(false)
    }
  }

  // Return the proper MUI elements
  return (
    <Paper
      data-testid="connectMainPanel"
      elevation={5}
      className={`${paperRoot} ${hidden ? panelHidden : (mouseIsOver ? panelExpanded : panelRetracted)}`}
      onMouseEnter={() => { setMouseIsOver(true); cancelHide() }}
      onMouseLeave={() => { setMouseIsOver(false); hide(false) }}
    >
      <PanelTitle title='Karuna Connect' arrow='right' onClose={() => { hide(true) }} />
      <div className={boxStyle}>
        <main className={contentStyle}>
          <ConnectMainContent />
        </main>
      </div>
    </Paper>
  )
}

ConnectMainPanel.propTypes = {
  /** Is the panel hidden */
  hidden: PropTypes.bool,

  /** Callback function when the panel should be hidden */
  onHide: PropTypes.func,

  /** Milliseconds to wait before hiding the panel */
  waitToHide: PropTypes.number
}

ConnectMainPanel.defaultProps = {
  hidden: false,
  onHide: null,
  waitToHide: 3000
}
