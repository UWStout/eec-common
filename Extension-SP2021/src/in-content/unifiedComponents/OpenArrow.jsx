import React from 'react'
import PropTypes from 'prop-types'

import { makeStyles } from '@material-ui/core/styles'
import Fade from '@material-ui/core/Fade'

import { KeyboardArrowRight, DoubleArrow } from '@material-ui/icons'

const useStyles = makeStyles((theme) => ({
  flipped: {
    transform: 'scale(-1, 1) translateY(-50%)',
    position: 'absolute',
    top: '50%'
  },
  standard: {
    transform: 'translateY(-50%)',
    position: 'absolute',
    top: '50%'
  }
}))

export default function OpenArrow (props) {
  const { showDouble, flipped } = props
  const classes = useStyles()

  return (
    <React.Fragment>
      <Fade in={showDouble}>
        <DoubleArrow className={flipped ? classes.flipped : classes.standard} />
      </Fade>
      <Fade in={!showDouble}>
        <KeyboardArrowRight className={flipped ? classes.flipped : classes.standard} />
      </Fade>
    </React.Fragment>
  )
}

OpenArrow.propTypes = {
  showDouble: PropTypes.bool,
  flipped: PropTypes.bool
}

OpenArrow.defaultProps = {
  showDouble: false,
  flipped: false
}
