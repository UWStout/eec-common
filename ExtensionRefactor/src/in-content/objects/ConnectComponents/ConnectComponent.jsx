import React, { useState, useEffect } from 'react'
import PropTypes from 'prop-types'

import ConnectForm from './ConnectForm.jsx'
import ConnectPanelButton from './ConnectPanelButton.jsx'

import { backgroundMessage } from '../AJAXHelper.js'

// The sidebar Karuna Connect object
export default function ConnectComponent (props) {
  // Form mode state (history, mood form open/closed)
  const [formOpen, updateFormOpen] = useState(false)
  const [historyFormOpen, updateHistoryFormOpen] = useState(false)

  // User/Karuna data state
  const [emojiList, updateEmojiList] = useState([])
  const [userStatus, updateUserStatus] = useState(null)
  const [privacy, updatePrivacy] = useState({ private: true, prompt: true })
  const [loginTrigger, updateLoginTrigger] = useState(0)

  // Initialize the privacy preferences
  useEffect(() => {
    backgroundMessage(
      { type: 'read', key: 'privacy' },
      'Failed to read privacy preferences: ',
      (newPrivacy) => {
        updatePrivacy({
          private: (newPrivacy?.private === undefined ? true : newPrivacy.private),
          prompt: (newPrivacy?.prompt === undefined ? true : newPrivacy.prompt)
        })
      }
    )
  }, [])

  // Initialize the emoji list
  useEffect(() => {
    // Send ajax request for data via background script
    backgroundMessage(
      { type: 'ajax-getEmojiList', loginTrigger },
      'Emoji Retrieval failed: ',
      (data) => { updateEmojiList(data) }
    )
  }, [loginTrigger])

  // Synchronize user state
  const getLatestUserStatus = () => {
    console.log('[[REACT]] Retrieving last user status')
    backgroundMessage(
      { type: 'ajax-getUserStatus' },
      'Retrieving current user status failed: ',
      (currentUserStatus) => {
        console.log('[[REACT]] User status retrieved:')
        console.log(currentUserStatus)
        if (props.emitter) { props.emitter.emit('userStatusChanged', currentUserStatus) }
        updateUserStatus(currentUserStatus)
      }
    )
  }

  // Set new user mood (triggers a userState update)
  const setUserMood = async (affectID, privacy) => {
    console.log('[[REACT]] Updating user mood')
    backgroundMessage(
      { type: 'ajax-setUserAffect', affectID, privacy },
      'Setting mood failed: ', () => {
        console.log('[[REACT]] user mood updated')
        getLatestUserStatus()
      }
    )
  }

  // Set new user mood (triggers a userState update)
  const setCollaboration = (newCollaboration) => {
    console.log('[[REACT]] Updating user collaboration status')
    backgroundMessage(
      { type: 'ajax-setCollaboration', collaboration: newCollaboration },
      'Setting collaboration failed: ', () => {
        console.log('[[REACT]] User collaboration status updated')
        getLatestUserStatus()
      }
    )
  }

  // Trigger first retrieval of user state
  useEffect(() => { getLatestUserStatus() }, [])

  // Listen for a login
  if (props.emitter) {
    props.emitter.on('login', () => {
      updateLoginTrigger(loginTrigger + 1)
      getLatestUserStatus()
    })
  }

  // Opens and closes main menu
  const handleClick = (e) => {
    const newFormOpen = !formOpen
    updateFormOpen(newFormOpen)
  }

  // Tracks history panel opening for button display
  const handleHistoryClick = (e) => {
    const newHistoryFormOpen = !historyFormOpen
    updateHistoryFormOpen(newHistoryFormOpen)
  }

  const handleHistoryBackClick = () => {
    updateHistoryFormOpen(false)
    updateFormOpen(true)
  }

  // Tracks state of each panel and displays karuna button if neither panel is open
  const PanelOpenButton = !formOpen && !historyFormOpen ? <ConnectPanelButton onClick={handleClick} /> : null

  return (
    <React.Fragment>
      {PanelOpenButton}
      <ConnectForm userStatus={userStatus} emojiList={emojiList} privacy={privacy} opened={formOpen}
        handleAffectChange={setUserMood} handleCollaborationChange={setCollaboration} handleClose={handleClick}
        handleHistoryFormOpen={handleHistoryClick} handleHistoryFormBack={handleHistoryBackClick} />
    </React.Fragment>
  )
}

ConnectComponent.propTypes = {
  emitter: PropTypes.object
}
