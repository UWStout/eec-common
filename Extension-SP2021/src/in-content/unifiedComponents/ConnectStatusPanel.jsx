import React from 'react'
import PropTypes from 'prop-types'

import { Paper } from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'

import OpenArrow from './OpenArrow.jsx'

const useStyles = makeStyles((theme) => ({
  // Styling of the root paper element
  paperRoot: {
    // Margins and sizing
    margin: theme.spacing(1),
    width: theme.spacing(16),
    height: theme.spacing(16),

    // Position element in window
    position: 'absolute',
    top: theme.spacing(40),

    // Animate changes in the 'right' property
    transition: theme.transitions.create(
      ['right'], { duration: theme.transitions.duration.standard }
    )
  },

  // Style when the panel is fully retracted
  panelRetracted: {
    right: -theme.spacing(13)
  },

  // Style when the panel is fully expanded
  panelExpanded: {
    right: -theme.spacing(8)
  },

  // Style when the panel is hidden
  panelHidden: {
    right: -theme.spacing(18)
  }
}))

/**
 * A small panel with an arrow that expands when hovered to show the user's status.
 * Will transition to the main panel when clicked.
 *
 * @param {object} props Properties for the component (See propTypes)
 * @returns {React.Element} The element to render for this component
 */
export default function ConnectStatusPanel (props) {
  // Deconstruct props and style class names
  const { hidden, onHide } = props
  const { paperRoot, panelRetracted, panelExpanded, panelHidden } = useStyles()

  // Is the mouse over this component
  const [mouseIsOver, setMouseIsOver] = React.useState(false)

  // Function to run when the component is clicked
  const clickCallback = () => {
    setMouseIsOver(false)
    if (onHide) { onHide() }
  }

  // Return the proper MUI elements
  return (
    <Paper
      elevation={3}
      className={`${paperRoot} ${hidden ? panelHidden : (mouseIsOver ? panelExpanded : panelRetracted)}`}
      onMouseEnter={() => { setMouseIsOver(true) }}
      onMouseLeave={() => { setMouseIsOver(false) }}
      onClick={clickCallback}
    >
      <OpenArrow showDouble={mouseIsOver} flipped />
    </Paper>
  )
}

ConnectStatusPanel.propTypes = {
  /** Is the panel hidden */
  hidden: PropTypes.bool,

  /** Callback function when the panel should be hidden */
  onHide: PropTypes.func
}

ConnectStatusPanel.defaultProps = {
  hidden: false,
  onHide: null
}
