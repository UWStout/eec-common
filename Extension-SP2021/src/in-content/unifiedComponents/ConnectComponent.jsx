/* global EventEmitter3 */

import React, { useState, useEffect } from 'react'
import PropTypes from 'prop-types'

import { makeStyles } from '@material-ui/core/styles'

import ConnectStatusPanel from './ConnectStatusPanel.jsx'
import ConnectMainPanel from './ConnectMainPanel.jsx'

import * as HELPER from './backgroundHelper.js'

// Colorful logger
import { makeLogger } from '../../util/Logger.js'
const LOG = makeLogger('CONNECT Component', 'lavender', 'black')

const useStyles = makeStyles((theme) => ({
  // Styling of the root paper element
  rootDiscord: {
    // Sizing to fill the page
    position: 'relative',
    width: '100%',
    height: '100%'

    // Set to be in front of all the other elements
    // zIndex: 100
  },
  rootTeams: {
    // Sizing to fill the page
    position: 'relative',
    width: '100%',
    height: '100%',

    // Ensure it is at the top of page (not the bottom)
    top: '-100vh'

    // Set to be in front of all the other elements
    // zIndex: 100

  }
}))

// The sidebar Karuna Connect object
export default function ConnectComponent ({ context, emitter }) {
  // Deconstruct props and style class names
  const { rootDiscord, rootTeams } = useStyles()

  // Is the mouse over this component
  const [mainPanelOpen, setMainPanelOpen] = useState(false)

  // Data shared throughout the connect panel is managed here
  const [emojiList, setEmojiList] = useState([])
  const [moodHistoryList, setMoodHistoryList] = useState([])
  const [currentStatus, setCurrentStatus] = useState(null)
  const [affectPrivacy, setAffectPrivacy] = useState(null)

  // Functions to retrieve state asynchronously
  const getEmojiList = async () => {
    try {
      const emojisFromServer = await HELPER.retrieveAffectList(context)
      // Filter out inactive emojis (old ones that have been removed)
      const filteredEmojis = emojisFromServer.filter((current) => (current.active))
      LOG('New Emoji List', filteredEmojis)
      setEmojiList(filteredEmojis)
    } catch (err) {
      LOG.error('Failed to retrieve emoji list', err)
    }
  }

  const getMoodHistoryList = async () => {
    try {
      const moodHistoryFromServer = await HELPER.retrieveAffectHistoryList(context)
      // get only the 10 most recent moods from the affect list
      // and make an array of only the affectIDs
      const mostRecentMoodsWithDuplicates = moodHistoryFromServer
        .map((item) => {
          return item.affectID
        })

      const mostRecentMoods = [...new Set(mostRecentMoodsWithDuplicates)].splice(0, 10)

      LOG('New Mood History List', mostRecentMoods)
      setMoodHistoryList(mostRecentMoods)
    } catch (err) {
      LOG.error('Failed to retrieve mood history list', err)
    }
  }

  const getPrivacy = async () => {
    try {
      const privacyFromStorage = await HELPER.retrieveMoodPrivacy(context)
      LOG('New Affect Privacy', privacyFromStorage)
      setAffectPrivacy(privacyFromStorage)
    } catch (err) {
      LOG.error('Failed to retrieve affect privacy', err)
    }
  }

  const updatePrivacy = async (newPrivacy) => {
    if (!newPrivacy) {
      LOG.error('(WARNING) Refusing to set privacy to null/undefined')
      return
    }

    try {
      await HELPER.setMoodPrivacy(newPrivacy, context)
      LOG('Affect Privacy Updated', newPrivacy)
      setAffectPrivacy(newPrivacy)
    } catch (err) {
      LOG.error('Failed to update affect privacy', err)
    }
  }

  const getCurrentStatus = async () => {
    try {
      const currentStatusFromServer = await HELPER.retrieveUserStatus(context)
      LOG('New Users Status', currentStatusFromServer)
      setCurrentStatus(currentStatusFromServer)
    } catch (err) {
      LOG.error('Failed to retrieve user status', err)
    }
  }

  const updateCurrentAffect = async (newAffectID, privacy = true) => {
    if (!newAffectID) {
      LOG.error('(WARNING) Refusing to set current mood to null/undefined')
      return
    }

    try {
      await HELPER.updateCurrentAffect(newAffectID, privacy, context)
      LOG('Current Mood Updated', newAffectID)
      await getCurrentStatus()
    } catch (err) {
      LOG.error('Failed to update current affect', err)
    }
  }

  // Retrieve initial values for all state
  useEffect(() => {
    getEmojiList()
    getMoodHistoryList()
    getPrivacy()
    getCurrentStatus()
  }, [])

  // Main render
  return (
    <div className={(context === 'msTeams' ? rootTeams : rootDiscord)}>
      <ConnectStatusPanel
        hidden={mainPanelOpen}
        onHide={() => { setMainPanelOpen(true) }}
        currentStatus={currentStatus}
        affectPrivacy={affectPrivacy}
      />
      <ConnectMainPanel
        hidden={!mainPanelOpen}
        onHide={() => { setMainPanelOpen(false) }}
        emojiList={emojiList}
        recentList={moodHistoryList}
        currentStatus={currentStatus}
        affectPrivacy={affectPrivacy}
        updateCurrentAffect={updateCurrentAffect}
        updatePrivacy={updatePrivacy}
      />
    </div>
  )
}

ConnectComponent.propTypes = {
  emitter: PropTypes.instanceOf(EventEmitter3).isRequired,
  context: PropTypes.string.isRequired
}
