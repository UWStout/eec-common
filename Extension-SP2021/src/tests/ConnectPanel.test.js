/**
 * @jest-environment jsdom
 */

import React from 'react'
import { 
  render, 
  fireEvent,
  screen,
  cleanup
} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { toBeVisible } from '@testing-library/jest-dom/matchers'

import OpenArrow from '../in-content/unifiedComponents/OpenArrow.jsx'
import ConnectStatusPanel from '../in-content/unifiedComponents/ConnectStatusPanel.jsx'
import ConnectMainPanel from '../in-content/unifiedComponents/ConnectMainPanel.jsx'
import ConnectComponent from '../in-content/unifiedComponents/ConnectComponent.jsx'

expect.extend({ toBeVisible })

describe('ConnectPanel', () => {
  afterEach(cleanup)

  describe('ConnectStatusPanel', () => {
    afterEach(cleanup)

    it('shows up', () => {
      const { getByTestId } = render(<ConnectStatusPanel hidden={true} />)
      const connectStatusPanel = getByTestId("connectStatusPanel")
      expect(connectStatusPanel).toBeInTheDocument()
    })
  
    it('shows a single arrow on load', () => {
      const { getByTestId } = render(<ConnectStatusPanel hidden={true} />)
      // the panel has a single arrow
      const singleArrow = getByTestId('keyboardArrowRight')
      expect(singleArrow).toBeInTheDocument()
    })
  
    it('shows double arrow on hover of single arrow', () => {
      const { getByTestId } = render(<ConnectStatusPanel hidden={true} />)
      
      // the panel has a single arrow
      const singleArrow = getByTestId('keyboardArrowRight')
      expect(singleArrow).toBeInTheDocument()
  
      // on mouseOver, a double arrow is shown
      fireEvent.mouseOver(singleArrow);
      const doubleArrow = getByTestId('doubleArrow')
      expect(doubleArrow).toBeInTheDocument()
    })

    it('shows double arrow on hover of status panel', () => {
      const { getByTestId } = render(<ConnectStatusPanel hidden={true} />)
      const connectStatusPanel = getByTestId("connectStatusPanel")
      // on mouseOver, a double arrow is shown
      fireEvent.mouseOver(connectStatusPanel);
      const doubleArrow = getByTestId('doubleArrow')
      expect(doubleArrow).toBeInTheDocument()
    })

    it('expands on hover', () => {
      const { getByTestId } = render(<ConnectStatusPanel hidden={true} />)
      const connectStatusPanel = getByTestId("connectStatusPanel")
      // on mouseOver, panel expands
      fireEvent.mouseOver(connectStatusPanel)
      expect(connectStatusPanel).toHaveStyle('right: -theme.spacing(8)')
    })

    it.todo('hides when mouse moves away')
  })

  describe('ConnectMainPanel', () => {
    it('shows up', () => {
      const { getByTestId } = render(<ConnectMainPanel hidden={false} />)
      const mainPanel = getByTestId("connectMainPanel")
      expect(mainPanel).toBeInTheDocument()
  
    })
    it('shows title', () => {
      render(<ConnectMainPanel hidden={false} />)
      const title = screen.getByRole('heading', { name: 'title' })
      expect(title).toBeInTheDocument()
    })

    it('shows main content', () => {
      render(<ConnectMainPanel hidden={false} />)
      const paragraph = screen.getByRole('main')
      expect(paragraph).toBeInTheDocument()
    })

    it('shows back arrow', () => {
      render(<ConnectMainPanel hidden={false} />)
      const closeButton = screen.getByRole("button", { name: 'close panel' })
      expect(closeButton).toBeInTheDocument()
    })
  })

  it('shows ConnectStatusPanel on load', () => {
    const { getByTestId } = render(<ConnectComponent />)
    
    // the panel with the single arrow on middle right of the screen shows up
    const connectStatusPanel = getByTestId("connectStatusPanel")
    expect(connectStatusPanel).toBeInTheDocument()
  })

  it('shows ConnectMainPanel on click', () => {
    const { getByTestId } = render(<ConnectComponent />)
  
    // Connect status panel shows up
    const connectStatusPanel = getByTestId("connectStatusPanel")
    
    // clicking the status panel brings up the main panel
    fireEvent.click(connectStatusPanel)
    const mainPanel = getByTestId("connectMainPanel")
    expect(mainPanel).toBeInTheDocument()

  })

  it('goes back to ConnectStatusPanel when ConnectMainPanel\'s back arrow is clicked', () => {
    const { getByTestId } = render(<ConnectComponent />)
  
    // Connect status panel shows up
    const statusPanel = getByTestId("connectStatusPanel")
    
    // clicking the status panel brings up the main panel
    fireEvent.click(statusPanel)
    const mainPanel = getByTestId("connectMainPanel")
    
    // clicking back arrow on the main panel leads back to status panel
    const closeMainPanel = screen.getByRole("button", { name: 'close panel' })
    fireEvent.click(closeMainPanel)
    expect(statusPanel).toBeInTheDocument()
  })
})
