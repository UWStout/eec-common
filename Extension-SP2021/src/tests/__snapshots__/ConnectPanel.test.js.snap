// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectPanel ConnectStatusPanel expands on hover 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root makeStyles-paperRoot-25 makeStyles-panelExpanded-27 MuiPaper-elevation3 MuiPaper-rounded"
    data-testid="connectStatusPanel"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root makeStyles-flipped-29"
      data-testid="doubleArrow"
      focusable="false"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      viewBox="0 0 24 24"
    >
      <path
        d="M15.5 5H11l5 7-5 7h4.5l5-7z"
      />
      <path
        d="M8.5 5H4l5 7-5 7h4.5l5-7z"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root makeStyles-flipped-29"
      data-testid="keyboardArrowRight"
      focusable="false"
      style="opacity: 0; webkit-transition: opacity 195ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 195ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      viewBox="0 0 24 24"
    >
      <path
        d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`ConnectPanel ConnectStatusPanel hides when mouse moves away 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root makeStyles-paperRoot-31 makeStyles-panelRetracted-32 MuiPaper-elevation3 MuiPaper-rounded"
    data-testid="connectStatusPanel"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root makeStyles-flipped-35"
      data-testid="doubleArrow"
      focusable="false"
      style="opacity: 0; webkit-transition: opacity 195ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 195ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      viewBox="0 0 24 24"
    >
      <path
        d="M15.5 5H11l5 7-5 7h4.5l5-7z"
      />
      <path
        d="M8.5 5H4l5 7-5 7h4.5l5-7z"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root makeStyles-flipped-35"
      data-testid="keyboardArrowRight"
      focusable="false"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      viewBox="0 0 24 24"
    >
      <path
        d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`ConnectPanel ConnectStatusPanel shows up correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root makeStyles-paperRoot-1 makeStyles-panelRetracted-2 MuiPaper-elevation3 MuiPaper-rounded"
    data-testid="connectStatusPanel"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root makeStyles-flipped-5"
      data-testid="doubleArrow"
      focusable="false"
      style="opacity: 0; visibility: hidden;"
      viewBox="0 0 24 24"
    >
      <path
        d="M15.5 5H11l5 7-5 7h4.5l5-7z"
      />
      <path
        d="M8.5 5H4l5 7-5 7h4.5l5-7z"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root makeStyles-flipped-5"
      data-testid="keyboardArrowRight"
      focusable="false"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      viewBox="0 0 24 24"
    >
      <path
        d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
      />
    </svg>
  </div>
</DocumentFragment>
`;
